SimTracker/TrackAssociatorProducers/plugins/QuickTrackAssociatorByHitsImpl.cc

```diff
@@ -727,3 +748,13 @@ double QuickTrackAssociatorByHitsImpl::weightedNumberOfTrackHits(const Trajector
   }
   return sum;
 }
+
+const TrackingRecHit* QuickTrackAssociatorByHitsImpl::getMostProbableHit(const SiTrackerMultiRecHit*& mHit) const{
+  std::vector<const TrackingRecHit*> componenthits = mHit->recHits();
+  int idmostprobable = 0;
+  for (unsigned int i=0; i < mHit->weights().size(); ++i){
+    if(mHit->weight(i) > mHit->weight(idmostprobable)) idmostprobable=i;
+  }
+
+  return componenthits[idmostprobable];
+}
```

Comment:
Nothing here depends on the members of `QuickTrackAssociatorByHitsImpl`, so this can be moved as a free function (inside the anonymous namespace at the top).